include/vast/Dialect/HighLevel/HighLevelOps.td

```diff
@@ -1260,6 +1260,58 @@ class AlignOfExprOpBase< string mnemonic, list< Trait > traits = [] >
 def AlignOfExprOp          : AlignOfExprOpBase< "alignof.expr" >;
 def PreferredAlignOfExprOp : AlignOfExprOpBase< "preferred_alignof.expr" >;
 
+def OffsetOfNodeAttr
+  : HighLevel_Attr< "OffsetOfNode", "offset_of_node" > {
+    let parameters = (ins
+      "std::variant< unsigned int, ::mlir::StringAttr >":$value
+      //TODO base class specifier
+    );
+
+  let summary = "VAST expr offsetofnode ";
+  let description = [{ This attribute corresponds to OffsetOfExprNode used by Clang AST.
+                       It can containt either an index into the list of expressions(regions) that index arrays
+                       or a field identifier or base class specifier.
+  }];
+
+  let builders = [
+    AttrBuilder< (ins "::mlir::StringAttr":$attr), [{
```

Comment:
do you need these two constructors? I would expect variant to be implicitly contructed.